src/adapter/src/catalog.rs

```diff
@@ -1679,6 +1681,8 @@ pub struct Index {
     pub conn_id: Option<ConnectionId>,
     pub depends_on: Vec<GlobalId>,
     pub cluster_id: ClusterId,
+    pub custom_logical_compaction_window: Option<Duration>,
+    pub is_retained_metrics_relation: bool,
```

Comment:
Nit: It feels a bit strange to call indexes "relations". Maybe that's just me.

Reply:
We specifically do no consider indexes to be relations in the code: https://github.com/MaterializeInc/materialize/blob/a0714eb8c26268dbd9ed6b0f49d8ae19dc573842/src/adapter/src/catalog.rs#L2109-L2124